### PR TITLE
Create blank results

### PR DESCRIPTION
**Pull request by jmtsuji**
Refactored the code so that the script operates on a file-by-file basis.

### CODE REVIEW

File: Visualization/CreateBlankResults.py [hunk 1]
```diff
@@ -5,88 +5,151 @@
...
+# Uses input query_proteins FAA file to build a blank BLAST CSV table. Returns the table.
+def generate_blank_results(query_proteins):
+
+    blank_results_list = []
+
+    try:
+        handle = open(query_proteins, "rU")
+        fasta_entries = SeqIO.parse(handle, "fasta")
+        for entry in fasta_entries:
+            blank_results_list.append(
```
Comment (LeeBergstrand) [inline Visualization/CreateBlankResults.py]: You could probably use python CSV module for this. However, it's probably not worth importing in this use case.

Comment (LeeBergstrand) [reply Visualization/CreateBlankResults.py]: https://docs.python.org/3/library/csv.html

Comment (LeeBergstrand) [reply Visualization/CreateBlankResults.py]: So You Want To Write Your Own CSV code?:

http://thomasburette.com/blog/2014/05/25/so-you-want-to-write-your-own-CSV-code/

**Comment by LeeBergstrand** [comment]
@jmtsuji Looks good but I left a bunch of comments.

File: Visualization/CreateBlankResults.py [hunk 2]
```diff
@@ -5,88 +5,151 @@
...
+        if not filename.endswith(".csv"):
+            print(ts() + "WARNING: '" + filename + "' may not be a CSV file!")
+
+    elif extension == "faa":
+        if not filename.endswith(".faa"):
+            print(ts() + "WARNING: '" + filename + "' may not be an amino acid FASTA file!")
+
+    else:
+        print(ts() + "ERROR: extension must be 'csv' or 'faa'. Exiting...")
+        exit(1)
```
[the remark this answers is not in the document]
Comment (jmtsuji) [reply Visualization/CreateBlankResults.py]: Addressed -- thanks for the FYI

File: Visualization/CreateBlankResults.py [hunk 3]
```diff
@@ -5,88 +5,151 @@
...
+    elif os.stat(input_csv).st_size > 0:
+        return False
+
+    else:
+        print(ts() + "ERROR: the input file appears to neither be empty nor non-empty. Exiting...")
+        exit(1)
+
+
+# Checks whether or not the provided file has the expected extension. Throws warning if not. No return.
+def file_extension_check(filename, extension):
```
[the remark this answers is not in the document]
Comment (jmtsuji) [reply Visualization/CreateBlankResults.py]: Done

File: Visualization/CreateBlankResults.py [hunk 4]
```diff
@@ -5,88 +5,151 @@
...
+    current_time = datetime.datetime.utcnow()
+    time_formatted = current_time.strftime("%a %b %d %H:%M:%S UTC %Y")
+
+    timestamp = "[ " + time_formatted + " ]: "
+
+    return(timestamp)
+
+
+# Checks whether or not the provided input_csv is an empty file. Returns logical.
+def check_if_input_CSV_is_empty(input_csv):
```
[the remark this answers is not in the document]
Comment (jmtsuji) [reply Visualization/CreateBlankResults.py]: Done

File: Visualization/CreateBlankResults.py [hunk 5]
```diff
@@ -5,88 +5,151 @@
...
-    exit(1)
-
-FakeResultsOut = "\n".join(FakeResults)
-
-print(">> Writing Fake Results...")
-
-for x in filesToReplaceList:
-    replacementFileName = x + ".csv"
+# Gives a timestamp (string) for help with logging
+def ts():
```
[the remark this answers is not in the document]
Comment (jmtsuji) [reply Visualization/CreateBlankResults.py]: @LeeBergstrand  Added logger -- my first time, cool. Does the setup code (top of main) look okay?

[the remark this answers is not in the document]
Comment (jmtsuji) [reply Visualization/CreateBlankResults.py]: Removed in place of logger

**Comment by jmtsuji** [comment]
@LeeBergstrand How does this look now?

File: Visualization/CreateBlankResults.py [hunk 6]
```diff
@@ -5,88 +5,109 @@
...
+
+# Writes a blank_results table to a file with name new_blast_file. No return.
+def write_blank_results(blank_results, new_blast_file):
+    with open(new_blast_file, "w") as writeFile:
+        writeFile.write(blank_results)
+
+
+def main(args):
+    # Set up the logger
+    logging.basicConfig(level=logging.DEBUG, format="[ %(asctime)s UTC ]: %(levelname)s: %(module)s: %(message)s")
```
Comment (LeeBergstrand) [inline Visualization/CreateBlankResults.py]: @jmtsuji In the future we may want to make the logging level toggleable via a CLI flag.

File: Visualization/CreateBlankResults.py [hunk 7]
```diff
@@ -5,88 +5,109 @@
...
+    logging.Formatter.converter = time.gmtime
+    logger = logging.getLogger(__name__)  # Not sure if this is needed
```
Comment (LeeBergstrand) [inline Visualization/CreateBlankResults.py]: I think this is needed. Remove the comment.

**Comment by LeeBergstrand** [comment]
@jmtsuji We may want to move this over to ```argsparse``` do you want to do that now or don't worry about it?

File: Visualization/CreateBlankResults.py (hunk 7 above)
Comment (jmtsuji) [reply Visualization/CreateBlankResults.py]: Done

**Comment by jmtsuji** [comment]
@LeeBergstrand I moved this over to argsparse. Logging levels could now be specified via flags at some future date (but maybe not needed at this stage?)

I also tested the code. Needed to move the logger setup outside main to have logger work globally for all functions. Had to make another minor fix as well. Seems to work now.

Thoughts?

**Comment by LeeBergstrand** [comment]
> Logging levels could now be specified via flags at some future date (but maybe not needed at this stage?). (Could potentially add a --verbose flag as in https://stackoverflow.com/a/20663028)

We could leave this for now. Let's merge this in.

**Pull request merged**